nixos/tests/nixos-rebuild-specialisations.nix

```diff
@@ -14,22 +14,11 @@ import ./make-test-python.nix ({ pkgs, ... }: {
         connect-timeout = 1;
       };
 
-      system.extraDependencies = with pkgs; [
-        curl
-        desktop-file-utils
-        docbook5
-        docbook_xsl_ns
-        grub2
-        kmod.dev
-        libarchive
-        libarchive.dev
-        libxml2.bin
-        libxslt.bin
-        python3Minimal
-        shared-mime-info
-        stdenv
-        sudo
-        xorg.lndir
+      system.includeBuildDependencies = true;
```

Comment:
Wouldn't this make the closure explode? :-(

I mean I dream of using this everywhere (while waiting for rebuildDependencyClosure which @JulienMalka is exploring).

Reply:
Not the build closure ;)

It balloons, but does not explode.
Right now the test itself explodes. Pieces everywhere.

Reply:
> @JulienMalka

Awesome!

Added some thoughts here https://github.com/NixOS/nixpkgs/issues/180529#issuecomment-1585732536

Reply:
So realistically I think our priorities should be

1. maintainable tests
2. working tests
3. fast tests

This PR doesn't make (3) harder, because it's easy enough to find tests that do `includeBuildDependencies = true`.
Meanwhile it fixes (1) and (2).

Reply:
I'm on board with the "correct by design".